maestro-domain/src/main/java/bio/overture/maestro/domain/api/DefaultIndexer.java

```diff
@@ -106,33 +142,56 @@
     }
 
     @Override
-    public Mono<IndexResult> indexStudy(@NonNull IndexStudyCommand indexStudyCommand) {
-        log.trace("in indexStudy, args: {} ", indexStudyCommand);
-        return tryGetStudyRepository(indexStudyCommand.getRepositoryCode())
-            .map(filesRepository -> toStudyAndRepositoryTuple(indexStudyCommand, filesRepository))
-            .flatMap(this::getStudyAnalysesDocuments)
-            .flatMap(this::batchUpsertFilesAndCollectFailures)
+    public Flux<IndexResult> indexStudy(@NonNull IndexStudyCommand command) {
+        List <Mono<IndexResult>> monos = new ArrayList<>();
```

Comment:
this probably means we will be fetching the study twice while we could only fetch it once.